src/SelectField/SelectField.js

```diff
@@ -0,0 +1,335 @@
+  /**
+   * The CSS class name of the `Menu` element.
+   */
+  menuClassName?: string,
+  /**
+   * Properties applied to the internal `<Menu />` component.
+   */
+  menuProps?: Object,
+  /**
+   * Name attribute of the `Input` element.
+   */
+  name?: string,
+  /** @ignore */
+  onBlur?: Function,
+  /**
+   * Callback function fired when a menu item is selected.
+   *
+   * @param {object} event TouchTap event targeting the menu item
+   * that was selected.
+   * @param {number} key The index of the selected menu item.
+   * @param {any} payload The `value` prop of the selected menu item.
+   */
+  onChange?: Function,
+  /** @ignore */
+  onClean?: Function,
+  /** @ignore */
+  onDirty?: Function,
+  /** @ignore */
+  onFocus?: Function,
+  placeholder?: string,
+  /**
+   * If `true`, the label is displayed as required.
+   */
+  required?: boolean,
+  /**
+   * Use that property to pass a ref callback to the root component.
+   */
+  rootRef?: Function,
+  /**
+   * Type attribute of the `Input` element. It should be a valid HTML5 input type.
+   */
+  type?: string,
+  /**
+   * The value of the `Input` element, required for a controlled component.
+   */
+  value?: string | number,
+};
+
+type AllProps = DefaultProps & Props;
+
+type State = {
+  anchorEl: Object,
+  ignoreFocusOnce: boolean,
+  open: boolean,
+  selectedIndex: number,
+};
+
+class SelectField extends Component<DefaultProps, AllProps, State> {
+  props: AllProps;
+  static defaultProps = {
+    compareFunction: (currentValue, itemValue) => currentValue === itemValue,
+    disabled: false,
+  };
+  state = {
+    anchorEl: undefined,
+    ignoreFocusOnce: false,
+    open: false,
+    selectedIndex: undefined,
+  };
+
+  handleMouseDown = event => event.preventDefault();
+
+  handleKeyDown = event => {
+    if (OPEN_MENU_KEYS.includes(keycode(event))) {
+      event.preventDefault();
+      this.setState({ open: true, anchorEl: event.currentTarget });
+    }
+  };
+
+  handleClick = event => {
+    event.currentTarget.focus();
+    this.setState({ open: true, anchorEl: event.currentTarget });
+  };
+
+  handleRequestClose = () => this.setState({ open: false });
+
+  handleSelectFocus = event => {
+    if (this.state.ignoreFocusOnce) {
+      event.stopPropagation();
+      this.setState({ ignoreFocusOnce: false });
+    }
+  };
+
+  handleSelectBlur = event => {
+    if (this.state.open) {
+      event.stopPropagation();
+    }
+  };
+
+  handleItemClick = (event, index, value) => {
+    event.persist();
+    this.setState(
+      {
+        open: false,
+        ignoreFocusOnce: true,
+      },
+      () => {
+        if (this.props.onChange) {
+          this.props.onChange(event, index, value);
+        }
+      },
+    );
+  };
+
+  render() {
+    const {
+      children,
+      autoFocus,
+      className,
+      compareFunction,
+      defaultValue,
+      disabled,
+      error,
+      hideLabel,
+      id,
+      inputClassName,
+      InputClassName,
```

Comment:
It's a bit confusing to have `inputClassName` and `InputClassName` here, Though I figured out from code why it's so, I'd suggest to find better naming, e.g. `inputClassName` and `innerInputClassName`, even `inputInputClassName` feels better to me, because it's so easy to make a mistake with case-sensitivity. Also why just don't allow user to put inner input props into `inputProps`, I mean inputProps: { inputProps: ..., inputClassName } and let only outer `Input` props on the top level?

Reply:
@kybarg Thanks for maintaining standards - if the consensus is to change this, best to do so consistently across all affected components.